stylis.js

```diff
@@ -1566,20 +1566,14 @@
 				break
 			}
 			default: {
-				switch (plugin.constructor) {
-					case Array: {
-						for (var i = 0, length = plugin.length; i < length; ++i) {
-							use(plugin[i])
-						}
-						break
-					}
-					case Function: {
-						plugins[plugged++] = plugin
-						break
-					}
-					case Boolean: {
-						unkwn = !!plugin|0
+				if (Object.prototype.toString.call(plugin) === '[object Array]') {
```

Comment:
Is there any reason you used this instead of typeof === 'object' and `Array.isArray`?

Reply:
The only reason that I didn't use `Array.isArray` is that that function was added to the JS spec somewhat recently and I wasn't sure if you want to be backwards compatible with older things. If you think that it is OK to use it I'd be happy to update the PR!

Reply:
Sure lets switch on `typeof` clauses and branch to the array branch if the 'object' has a `.length` > -1.

Reply:
Apologies for the slow follow up. Is what you had in mind 8a1887f06645d4e046ffd0e11ae308740d6901e2? Doing a `switch (typeof plugin)` is a little bit annoying because
```js
> typeof false === typeof new Boolean(false)
false
```

Reply:
Yes something along those lines. i.e `if (typeof === 'function') {...function branch} else if (.length > -1) {...array branch} else {...boolean branch}`.

Reply:
Ok! Do you want me to change it to exactly that or is the latest commit that I pushed ok? I don't have a preference :slightly_smiling_face:

Reply:
Sorry for the delay, i wrote a comment last week but didn't notice that it didn't get sent. 

Yes and a else if typeof 'boolean' or an default else branch should also suffix for the boolean case.

Reply:
> Sorry for the delay, i wrote a comment last week but didn't notice that it didn't get sent.

No worries!

> Yes and a else if typeof 'boolean' or an default else branch should also suffix for the boolean case.

Ok! I think I'll go for just using a default else branch for the boolean case. One annoying thing with boolean values in javascript is that
```javascript
> typeof new Boolean(true)
"object"
```
which is why I used equality checks for `true` and `false` (since they are fast) and then a `instanceof Boolean` check to catch 'boolean objects' too. I'll push a commit with those changes soon 😄

Reply:
I pushed a new commit now, 5e7ae10!

Reply:
@thysultan did you have a chance to look at my latest commit yet? 🙂